File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const realLost = lost.filter(l => !reserve.includes(l)).sort((a, b) => a - b);
    const realRes = reserve.filter(r => !lost.includes(r)).sort((a, b) => a - b);
    
    for (let r of realRes) {
        // 앞 번호
        const frontIdx = realLost.indexOf(r - 1);
        if (frontIdx !== -1) {
            realLost.splice(frontIdx, 1);
            continue;
        }
        
        // 뒷 번호
        const afterIdx = realLost.indexOf(r + 1);
        if (afterIdx !== -1) {
            realLost.splice(afterIdx, 1);
        }
    }
    return n - realLost.length;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    overlap = set(lost) & set(reserve)
    
    lost = sorted(set(lost)-overlap)
    reserve = sorted(set(reserve)-overlap)
    
    for r in reserve:
        if r-1 in lost:
            lost.remove(r-1)
        elif r+1 in lost:
            lost.remove(r+1)
    return n - len(lost)